dev/include/hw/multiboot.h

```diff
@@ -172,5 +223,82 @@ enum {
     MULTIBOOT_FRAMEBUFFER_TYPE_EGA_TEXT = 2,
 };
 
+#if MULTIBOOT2_SUPPORT
+
+struct multiboot2_info {
+    uint32_t total_size;
+    uint32_t reserved;
+};
+
+struct multiboot2_tag {
+    uint32_t type;
+    uint32_t size;
+};
+
+struct multiboot2_tag_efi64 {
+    struct multiboot2_tag tag;
+    uint64_t pointer;
+};
+
+struct multiboot2_tag_mmap {
+    struct multiboot2_tag tag;
+    uint32_t entry_size;
+    uint32_t entry_version;
+};
+
+struct multiboot2_mmap_entry {
+    uint64_t addr;
+    uint64_t len;
+    uint32_t type;
+    uint32_t zero;
+};
+
+struct multiboot2_tag_framebuffer_common {
+    struct multiboot2_tag tag;
+    uint64_t framebuffer_addr;
+    uint32_t framebuffer_pitch;
+    uint32_t framebuffer_width;
+    uint32_t framebuffer_height;
+    uint8_t framebuffer_bpp;
+    uint8_t framebuffer_type;
+    uint16_t reserved;
+};
+
+struct multiboot2_tag_framebuffer {
+    struct multiboot2_tag_framebuffer_common common;
+    union {
+        struct {
+            uint16_t framebuffer_palette_num_colors;
+        };
```

Comment:
suggestion: can we name these enums something to disambiguate them at run time? ie, fb.rgb.* or fb.palette.*?